### PR TITLE
feat: [SRM-10018]: Initial draft MonitoredService template

### PR DESCRIPTION
**Pull request by deepesh-ui**
##### Screenshots:
Screenshot 2022-04-12 at 8 16 21 AM
Screenshot 2022-04-12 at 8 14 57 AM
Screenshot 2022-04-12 at 8 14 49 AM
Screenshot 2022-04-12 at 8 14 26 AM
Screenshot 2022-04-18 at 11 58 17 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.tsx [hunk 1]
```diff
@@ -134,33 +139,52 @@ function Service({
...
+          updateTemplate?.({
+            identifier,
+            name: 'Test Template',
+            orgIdentifier,
+            projectIdentifier,
+            spec: {
+              spec: { serviceRef: serviceRef, environmentRef: environmentRef },
+              timeout: '2m',
+              type: 'Http'
+            } as any,
```
Comment (diptiman-adak) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/Service.tsx:152]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9235241)
```suggestion
            } as unknown,
```

File: src/modules/85-cv/components/HarnessServiceAndEnvironment/components/EnvironmentSelectOrCreate/EnvironmentSelectOrCreate.tsx [hunk 2]
```diff
@@ -46,30 +49,52 @@ export function EnvironmentSelectOrCreate({
...
+          name={'environmentRef'}
+          disabled={disabled}
+          selectProps={{
+            items: environmentOptions
+          }}
+          allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME]}
+          value={item}
+          style={{ width: '400px' }}
+          expressions={expressions}
+          onChange={(val: any) => onSelectChange(val)}
```
Comment (diptiman-adak) [inline src/modules/85-cv/components/HarnessServiceAndEnvironment/components/EnvironmentSelectOrCreate/EnvironmentSelectOrCreate.tsx:84]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9235241)
```suggestion
          onChange={(val: unknown) => onSelectChange(val)}
```

File: src/modules/85-cv/components/HarnessServiceAndEnvironment/components/ServiceSelectOrCreate/ServiceSelectOrCreate.tsx [hunk 3]
```diff
@@ -69,27 +79,46 @@ export const ServiceSelectOrCreate: React.FC<ServiceSelectOrCreateProps> = props
...
+        <MultiTypeInput
+          name={name ?? 'service'}
+          disabled={props.disabled}
+          selectProps={{
+            items: selectOptions
+          }}
+          allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME]}
+          value={props.item}
+          style={{ width: '400px' }}
+          onChange={(val: any) => onSelectChange(val)}
```
Comment (diptiman-adak) [inline src/modules/85-cv/components/HarnessServiceAndEnvironment/components/ServiceSelectOrCreate/ServiceSelectOrCreate.tsx:106]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9235241)
```suggestion
          onChange={(val: unknown) => onSelectChange(val)}
```

File: src/modules/72-templates-library/utils/templatesUtils.ts [hunk 4]
```diff
@@ -40,41 +41,54 @@ export const getScopeBasedQueryParams = (
...
+      value: TemplateType.StepGroup,
+      disabled: true
+    },
+    {
+      label: getString('executionText'),
+      value: TemplateType.Execution,
+      disabled: true
+    }
+  ]
+  if (module !== 'cv') {
```
Comment (harshilbits) [inline src/modules/72-templates-library/utils/templatesUtils.ts:87]: Is there a better way of handling this, than adding a if condition here.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**